src/api/BaseApi.ts

```diff
@@ -0,0 +1,81 @@
+import http from '../classes/HTTPTransport';
+import env from '../utils/env';
+import { convertKeysToCamelCase } from '../utils/keysConverter';
+
+const defaultHeaders = {
+  'Content-type': 'application/json; charset=UTF-8',
+};
+
+interface IBaseApi {
+  baseUrl?: string,
+  path?: `/${string}`
+  headers?: Record<string, string>
+}
+
+class BaseApi {
+  private _http: typeof http;
+  private _baseUrl: string;
+  private _path: string;
+  private _headers: Record<string, string>;
+
+  constructor(config: IBaseApi = {}) {
+    this._http = http;
+    this._baseUrl = config.baseUrl || env.HOST_API || '';
+    this._path = config.path || '';
+    this._headers = config.headers || defaultHeaders;
+  }
+
+  private getPath() {
+    return `${this._baseUrl}${this._path}`;
+  }
+
+  private handleOptions(newOptions?: Record<any, any>) {
```

Comment:
Почему бы не написать string?
```suggestion
  private handleOptions(newOptions?: Record<string, string>) {
```